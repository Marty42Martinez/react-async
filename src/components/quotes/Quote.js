import React from 'react';
import PropTypes from 'prop-types';

function Quote({ quote }) {
  return (
    <section>
      <img src={quote.image}/>
      <p>{quote.character}</p>
      <p>{quote.quote}</p>
    </section>
  );
}

Quote.propTypes = {
  quote: PropTypes.shape({
    img: PropTypes.string,
    character: PropTypes.string,
    quote: PropTypes.quote
  }).isRequired
};


export default Quote;
