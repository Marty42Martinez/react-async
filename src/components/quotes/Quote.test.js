import React from 'react';
import { shallow } from 'enzyme';
import Quote from './Quote';

describe('Quote component', () => {
  it('renders a Quote', () => {
    const quote = {
      img: 'path/to/image',
      character: 'Bender',
      quote: 'Bite my shiny metal, ass'
    };

    const wrapper = shallow(<Quote quote={quote} />);
    expect(wrapper).toMatchSnapshot();
  });
});
