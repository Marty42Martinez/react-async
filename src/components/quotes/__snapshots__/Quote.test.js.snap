// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quote component renders a Quote 1`] = `
<section>
  <img
    src="path/to/image"
  />
  <p>
    Bender
  </p>
  <p>
    Bite my shiny metal, ass
  </p>
</section>
`;
